fairlearn/datasets/synthetic.py

```diff
@@ -0,0 +1,66 @@
+# Copyright (c) Microsoft Corporation and Fairlearn contributors.
+# Licensed under the MIT License.
+import numpy as np
+from sklearn.datasets import make_classification
+
+
+def make_synthetic_dataset(seed=12345):
+    """Create a synthetic dataset.
+
+    Parameters
+    ----------
+    seed : int, default=12345
+        The random number generator seed to use.
+
+    Returns
+    -------
+    (X, y, gender) : tuple of numpy.ndarray
+        X : ndarray
+            The generated samples.
+        y : ndarray
+            Labels for the binary classification.
+        gender : ndarray
+            The sensitive feature label.
+    """
+    rng = np.random.RandomState(seed=seed)
+
+    classification_kwargs = {
+        'n_features': 20,
+        'n_informative': 4,
+        'n_classes': 2,
+        'random_state': rng,
+    }
+
+    X_women, y_women = make_classification(
+        n_samples=500,
+        class_sep=1,
+        **classification_kwargs,
+    )
+
+    X_men, y_men = make_classification(
+        n_samples=500,
+        class_sep=2,
+        **classification_kwargs,
+    )
+
+    X_other, y_other = make_classification(
+        n_samples=500,
+        class_sep=0.5,
+        **classification_kwargs,
+    )
+
+    X_unspecified, y_unspecified = make_classification(
```

Comment:
#792 specifies the categories one should ideally provide to users. I'm not sure whether this needs to conform to that, but it's definitely something we should keep on our radar for a follow-up PR if not for this one. 
A related point that occurred to me was that the group sizes should probably be configurable. My initial thought was to have a dictionary where for each sensitive feature (gender, race, nationality, etc.) you can specify the group sizes ("male": 300, "female": 400, etc.) but then it's unclear how to map that onto individuals so that it still makes sense (e.g., should races be distributed equally among the genders?), so it's probably best to let the user specify that (?).
I do think this goes beyond the initial PR for sure, but if you have thoughts on how this could be achieved in a sensible way we should get the conversation started 🙂